lib/sles4sap_publiccloud.pm

```diff
@@ -78,9 +89,145 @@ sub get_promoted_hostname {
     return join("", @master);
 }
 
+=head2 get_hana_topology
+    get_hana_topology([hostname => $hostname]);
+
+    Parses command output, returns list of hashes containing values for each host.
+    If hostname defined, returns hash with values only for host specified.
+=cut
+
+sub get_hana_topology {
+    my ($self, %args) = @_;
+    my @topology;
+    my $hostname = $args{hostname};
+    my $cmd_out = $self->run_cmd(cmd => "SAPHanaSR-showAttr --format=script", quiet => 1);
+    record_info("cmd_out", $cmd_out);
+    my @all_parameters = map { if (/^Hosts/) { s,Hosts/,,; s,",,g; $_ } else { () } } split("\n", $cmd_out);
+    my @all_hosts = uniq map { (split("/", $_))[0] } @all_parameters;
+
+    for my $host (@all_hosts) {
+        my %host_parameters = map { my ($node, $parameter, $value) = split(/[\/=]/, $_);
+            if ($host eq $node) { ($parameter, $value) } else { () } } @all_parameters;
+        push(@topology, \%host_parameters);
+
+        if (defined($hostname) && $hostname eq $host) {
+            return \%host_parameters;
+        }
+    }
+
+    return \@topology;
+}
+
+=head2 is_hana_online
+    is_hana_online([timeout => 120, wait_for_start => 'false']);
+
+    Check if hana DB is online. Define 'wait_for_start' to wait for DB to start.
+
+=cut
+
+sub is_hana_online {
+    my ($self, %args) = @_;
+    my $wait_for_start = defined($args{wait_for_start}) ? 1 : 0;
```

Comment:
I see I have written it more complicated than should be. 

The idea is:
True = it will check db status until timeout or db is online
False = just return value of current DB status

Yes, thanks. I will use the example with changin L144 to:
```
return $db_status unless $wait_for_start;
```